src/wasm/wasm-binary.cpp

```diff
@@ -2963,29 +2961,16 @@ void WasmBinaryReader::pushExpression(Expression* curr) {
   if (type.isTuple()) {
     // Store tuple to local and push individual extracted values
     Builder builder(wasm);
-    // Non-nullable types require special handling as they cannot be stored to
-    // a local, so we may need to use a different local type than the original.
-    auto localType = type;
-    if (!wasm.features.hasGCNNLocals()) {
-      std::vector<Type> finalTypes;
-      for (auto t : type) {
-        if (t.isNonNullable()) {
-          t = Type(t.getHeapType(), Nullable);
-        }
-        finalTypes.push_back(t);
-      }
-      localType = Type(Tuple(finalTypes));
+    std::vector<Type> finalTypes;
+    for (auto t : type) {
+      finalTypes.push_back(t);
     }
     requireFunctionContext("pushExpression-tuple");
-    Index tuple = builder.addVar(currFunction, localType);
+    Index tuple = builder.addVar(currFunction, type);
     expressionStack.push_back(builder.makeLocalSet(tuple, curr));
-    for (Index i = 0; i < localType.size(); ++i) {
+    for (Index i = 0; i < type.size(); ++i) {
       Expression* value =
-        builder.makeTupleExtract(builder.makeLocalGet(tuple, localType), i);
-      if (localType[i] != type[i]) {
-        // We modified this to be nullable; undo that.
-        value = builder.makeRefAs(RefAsNonNull, value);
-      }
+        builder.makeTupleExtract(builder.makeLocalGet(tuple, type), i);
       expressionStack.push_back(value);
```

Comment:
I would fuzz this change carefully, as I'm not sure we always treat a tuple with a NN field as a NN "thing". That is, such a tuple local should be assigned before it is read, like a NN local.